### PR TITLE
dts: arm: move pinmux to dts

### PR DESCRIPTION
**Pull request by Kludentwo**
Signed-off-by: Nicolai Glud <nicolai.glud@prevas.dk>

### CODE REVIEW

**Comment by Kludentwo** [comment]
@galak I cannot get the property in pinmux.c.
I will have a look at it later, maybe you could give a pointer to how you access nested dt entries?

**Comment by Kludentwo** [comment]
Referencing:
#33262 and #26528

**Comment by galak** [comment]
@Kludentwo any thoughts on how to collaborate on this PR?

I've got a hack python script that will generate devicetrees for pin information based on the MCUX config data.

I'm happy to push changes to this PR if you are ok with that?

**Comment by Kludentwo** [comment]
Yeah, go ahead.

**Comment by Kludentwo** [comment]
most of the tests failing are about nxp,kinetis-pinmux.yaml being changed. Should we create an "old" config element for the other boards or do the same work in all the boards at the same time?

**Comment by galak** [comment]
> most of the tests failing are about nxp,kinetis-pinmux.yaml being changed. Should we create an "old" config element for the other boards or do the same work in all the boards at the same time?

we need to do the same work to the other boards.  I have a script that generates the pinmux .dtsi files based on the NXP config tool data.

Mostly, want to get @MaureenHelm's feedback on direction here, what things we need to tweak and get this agreed upon for one platform.  Will be easy enough to update the others once we have that.

For example, for now I've left the port reference in `nxp,kinetis-pins`, however if we keep the structure of these being defined per port, then its duplicated information as the parent is always the port, so I'd remove that and have `nxp,kinetis-pins` just be in the PIN and MUX value.

File: dts/arm/nxp/pinmux_nxp_kinetis.h [hunk 1]
```diff
@@ -23,10 +26,14 @@
...
-		mux = <k_mux>; \
-	}
+#define DT_PINCFG_FLAG(flag) flag;
+#define DT_PINCFG_FLAGS(...) \
+       MACRO_MAP_CAT(DT_PINCFG_FLAG __VA_OPT__(,) __VA_ARGS__)
+
+#define DT_NXP_PINMUX_DEFINE(k_name, k_port, k_pin, k_mux, ...) \
+        k_name##_##pt##k_port##k_pin: k_name##_##pt##k_port##k_pin { \
+               nxp,kinetis-pins = < &port##k_port k_pin k_mux >; \
+               DT_PINCFG_FLAGS(__VA_ARGS__) \
```
Comment (MaureenHelm) [inline dts/arm/nxp/pinmux_nxp_kinetis.h:36]: What is this for?

**Review by MaureenHelm — COMMENTED** [review]
Overall this looks really good. I agree the port reference in `nxp,kinetis-pins` seems redundant; can you think of a reason we would not want to define these per port?

We're not using them today, but how would we add other pin configuration fields like drive strength, slew rate, and passive filter enable (see PORTx_PCRn field descriptions in the HWRM)?

We might want to make the compat a little more specific, like `nxp,kinetis-port-pins`. I need to check if there parts in the Kinetis family that use different IP than the PORT module.

File: dts/arm/nxp/pinmux_nxp_kinetis.h (hunk 1 above)
Comment (galak) [reply dts/arm/nxp/pinmux_nxp_kinetis.h:36]: Its a hold over from non-generated dtsi files.

Comment (galak) [reply dts/arm/nxp/pinmux_nxp_kinetis.h:36]: Will remove in next iteration.

**Pull request closed without merging**